### PR TITLE
quic: Use name() instead of DebugString() 

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
quic: Use name() instead of DebugString() when checking that the QUIC transport socket is configured

Risk Level: Low
Testing: Existing
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by RyanTheOptimist** [comment]
/assign @alyssawilk 
I noticed this attempting to land Lite Proto support

**Review by alyssawilk — DISMISSED** [review]
definite improvement, thanks!

**Review by alyssawilk — DISMISSED** [review]

**Review by alyssawilk — COMMENTED** [review]
LG but let's add unit tests and a comment explanation of why this exception

**Review by alyssawilk — APPROVED** [review]

File: source/common/upstream/upstream_impl.cc [hunk 1]
```diff
@@ -1382,6 +1383,23 @@ ClusterInfoImpl::upstreamHttpProtocol(absl::optional<Http::Protocol> downstream_
...
 
+bool validateTransportSocketSupportsQuic(
+    const envoy::config::core::v3::TransportSocket& transport_socket) {
+  // The transport socket is valid for QUIC if it is either a QUIC transport socket,
+  // or if it is a QUIC transport socket wrapped in an HTTP/1.1 proxy socket.
+  if (transport_socket.name() == "envoy.transport_sockets.quic") {
+    return true;
+  }
+  if (transport_socket.name() != "envoy.transport_sockets.http_11_proxy" &&
+      transport_socket.name() != "envoy.transport_sockets.upstream_http_11_proxy") {
```
Comment (RyanTheOptimist) [inline source/common/upstream/upstream_impl.cc:1394]: @alyssawilk Looks like we also have an upstream version of the proxy transport socket.

Comment (alyssawilk) [reply source/common/upstream/upstream_impl.cc:1394]: Where?  I see the one in source/extensions/extensions_metadata.yaml but I'm not seeing the latter in E-M or Envoy with a quick grep around
/wait-any

Comment (RyanTheOptimist) [reply source/common/upstream/upstream_impl.cc:1394]: I was seeing failures from test/extensions/transport_sockets/http_11_proxy/connect_integration_test.cc without this change. Maybe we should change the transport_socket name in that test?

Comment (alyssawilk) [reply source/common/upstream/upstream_impl.cc:1394]: yeah, lets

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**